mlrun/builder.py

```diff
@@ -321,18 +321,12 @@ def build_image(
     image_target, secret_name = _resolve_image_target_and_registry_secret(
         image_target, registry, secret_name
     )
-    # TODO: currently requirements are not being passed to that method, this is due to the ImageBuilder class not having
-    #   requirements attribute in it, remove this comment when requirements attribute is being added to the class and
-    #   passed to the `build_image` method. Also `with_requirements` will have to be changed to set them to the
-    #   requirements attribute instead of transforming it right to commands
-    if isinstance(requirements, list):
+    if requirements and isinstance(requirements, list):
```

Comment:
I've set some users using set as their requirements, try to catch that too

Reply:
set is being handled in `with_requirements` as part of initializing a new list and appending to it.